oauthproxy.go

```diff
@@ -929,11 +929,19 @@ func (p *OAuthProxy) OAuthCallback(rw http.ResponseWriter, req *http.Request) {
 	}
 }
 
-// AuthenticateOnly checks whether the user is currently logged in
-func (p *OAuthProxy) AuthenticateOnly(rw http.ResponseWriter, req *http.Request) {
+// AuthOnly checks whether the user is currently logged in (both authentication
+// and optional authorization via `allowed_groups` querystring).
+func (p *OAuthProxy) AuthOnly(rw http.ResponseWriter, req *http.Request) {
 	session, err := p.getAuthenticatedSession(rw, req)
 	if err != nil {
-		http.Error(rw, "unauthorized request", http.StatusUnauthorized)
+		http.Error(rw, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		return
+	}
+
+	// Allow secondary group restrictions based on the `allowed_group` or
+	// `allowed_groups` querystring parameter
+	if !checkAllowedGroups(req, session) {
+		http.Error(rw, http.StatusText(http.StatusForbidden), http.StatusForbidden)
```

Comment:
I can see future contributors wanting to add more to this and we will end up with a really complicated chain here, might be better to wrap this in some authz function that just callls this same logic for now, so that we clearly have authn and authz separated in the logic, WDYT?

Reply:
> The plural in the configuration is going away in the long term, not sure how I feel about having this here or whether we should just have allowed_groups and be done with it

I'm all for simplifying 😄 

Which did you want to align on? One comma-delimited querystring of the groups (plural case). Or singular querstring case that can be specified multiple times?

Reply:
> I can see future contributors wanting to add more to this and we will end up with a really complicated chain here, might be better to wrap this in some authz function that just callls this same logic for now, so that we clearly have authn and authz separated in the logic, WDYT?

I'm up for making the change 👍  -- I'm having trouble visualizing the function/chain you are thinking to simplify this. Do you have a small pseudocode sample of what you had in mind?